docs/DeveloperGuide.md

```diff
@@ -202,6 +202,61 @@ detailed view:
   * Pros: More in-line with the purpose of the detailed view of showing only one book
   * Cons: More work has to be done to sync up the UI with the model.
   
+### General _bookmark_ Command Logic
+
+_bookmark_'s commands generally follow the same procedure. The following section will explain the 
+general flow of events for the following commands: 
+* `add`
+* `list`
+* `view`
+* `edit`
+* `delete` 
+
+The listed commands follow a similar sequence diagram as adding a book. These commands are 
+facilitated by `LogicManager`, `ModelManager`, `XYZCommandParser` and `XYZCommand` where XYZ refers to the specific command.
+eg. List, View, Edit etc. 
+
+`LogicManager` will handle the user input by first identifying the type of command (eg. Add, List, View, Edit etc.) and create a 
+`XYZCommandParser` object that will handle the remaining user input if any to produce a `XYZCommand`. 
+
+`XYZCommand` will then handle the specific action of the command and interact with the `ModelManager` to display the appropriate
+list of book(s). A `CommandResult` object which contains the return message 
+to the User will be created by `XYZCommand` and returned to the user.
```

Comment:
Instead of returned to the user, maybe use "passed to the Ui component to be displayed to the user"?